File: Assets/LuaScripts/_Test/DemoClass.lua
DemoClass = {}

ParentClass = BaseClass('ParentClass')
ParentClass.parentName = 'parent'
ParentClass.id = 10000

AClass = BaseClass('AClass', ParentClass)
AClass.base = ParentClass
AClass.aName = 'aClass'
function AClass:constructor()
    self.base.parentName = self.aName .. ' ' .. self.base.parentName
    self.base.id = CS.UnityEngine.Random.value * self.base.id
end

BClass = BaseClass('BClass', ParentClass)
BClass.base = ParentClass
BClass.bName = 'bClass'
function BClass:constructor()
    self.base.parentName = self.bName .. ' ' .. self.base.parentName
    self.base.id = CS.UnityEngine.Random.value * self.base.id
end

function DemoClass:Test()
    print('require democlass teset')
    AClass.New()
    BClass.New()

    print('aName:',AClass.aName,',parent:',AClass.base.parentName,',id:',AClass.base.id)
    print('bName:',BClass.bName,',parent:',BClass.base.parentName,',id:',BClass.base.id)

end

return DemoClass

File: Assets/_JustJump/Scripts/DemoClass.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class DemoClass
{

    public DemoClass()
    {
        AClass a = new AClass();
        BClass b = new BClass();
        
        Debug.Log(string.Format("aName:{0},parent:{1},id:{2}", a.aName, a.parentName,a.id));
        Debug.Log(string.Format("bName:{0},parent:{1},id:{2}", b.bName, b.parentName,b.id));
    }

}

public class ParentClass
{
    public string parentName="parent";
    public int id = 1000;
}

public class AClass: ParentClass
{
    public string aName = "aClass";

    public AClass()
    {
        parentName = aName + " " + parentName;
        id =(int)(Random.value * id);
    }
}

public class BClass:ParentClass
{
    public string bName = "bClass";

    public BClass()
    {
        parentName = bName + " " + parentName;
        id = (int)(Random.value * id);
    }
}

